### PR TITLE
Refactor rate limit handling and unify conversations pagination

### PR DESCRIPTION
**Pull request by hihats**
## Summary
レート制限の処理を一元化・強化し、Slack Conversations API呼び出し用の共有モジュールを追加し、テストカバレッジを向上させることで、Slack関連のPythonスクリプトに大幅な改善をもたらす。

This pull request introduces significant improvements to the Slack-related Python scripts by centralizing and enhancing rate limit handling, adding a shared module for Slack conversations API calls, and increasing test coverage. The most important changes are as follows:

**Rate Limit Handling Improvements:**

* Introduced a new shared module `slack_rate_limit.py` with an improved `handle_rate_limit` function that handles Slack API rate limits using the `Retry-After` header and exponential backoff, and is now imported and used across other modules instead of duplicating the logic. <a>[1]</a> <a>[2]</a> <a>[3]</a>

**Slack Conversations API Abstraction:**

* Added `slack_conversations.py`, a new shared module that abstracts and centralizes the logic for paginated and rate-limited calls to Slack's `conversations.history` and `conversations.replies` APIs, making it easier for scripts to fetch messages and thread replies.
* Fixed a bug in `fetch_thread_replies` where `is_first_page` was only set to `False` when replies were non-empty, causing the second page to incorrectly drop its first message when the first page returned an empty list.

**Bug Fixes and Consistency:**

* Updated `get_channel_last_message_time` in `inactive_channels.py` to always return a UTC-aware `datetime` object, ensuring consistent timezone handling.

**Testing Enhancements:**

* Added a comprehensive new test suite `tests/test_inactive_channels.py` to cover channel loading, message time retrieval, inactive channel checks, and result saving logic, improving reliability and maintainability.
* Added a comprehensive new test suite `tests/test_slack_conversations.py` covering cursor pagination (`has_more`/`next_cursor` handling), `skip_parent` behavior across single and multi-page scenarios, rate-limit retry logic, and edge cases such as empty responses and `ok=False`.
* Improved mock response construction in `tests/test_slack_search.py` for better test consistency and clarity.

### CODE REVIEW

File: slack_conversations.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+        messages = response.get("messages", [])
+        yield messages
+
+        if response.get("has_more") and response.get("response_metadata", {}).get("next_cursor"):
+            cursor = response["response_metadata"]["next_cursor"]
+            time.sleep(_TIER3_INTERVAL)
+        else:
+            break
+
```
[the remark this answers is not in the document]
Comment (hihats) [reply slack_conversations.py:51]: @copilot apply changes based on [this feedback](https://github.com/hihats/slackapp/pull/16#discussion_r3028800780)

File: tests/test_slack_search.py [hunk 2]
```diff
@@ -116,18 +116,20 @@ def test_non_ratelimit_error_raises_immediately(self):
 def _mock_search_response(matches, page=1, page_count=1):
     """search_messages API レスポンスのモックを生成"""
     resp = MagicMock()
-    resp.__getitem__ = lambda self, key: True if key == "ok" else None
-    resp.__bool__ = lambda self: True
-    resp.get = lambda key, default=None: {
+    data = {
+        "ok": True,
```
[the remark this answers is not in the document]
Comment (hihats) [reply tests/test_slack_search.py:119]: #9 で解消済み

File: tests/test_slack_search.py [hunk 3]
```diff
@@ -116,18 +116,20 @@ def test_non_ratelimit_error_raises_immediately(self):
...
                 "total_count": len(matches) * page_count,
             },
-        }
-    }.get(key, default)
+        },
+    }
+    resp.__getitem__ = lambda self, key: data[key]
+    resp.__bool__ = lambda self: True
+    resp.get = lambda key, default=None: data.get(key, default)
     return resp
```
[the remark this answers is not in the document]
Comment (hihats) [reply tests/test_slack_search.py:133]: #9 で解消済み

File: slack_conversations.py [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+
+def fetch_channel_history(
+    client: WebClient,
+    channel_id: str,
+    oldest: Optional[float] = None,
+    latest: Optional[float] = None,
+    limit: int = 200,
+) -> Generator[List[Dict], None, None]:
+    """conversations.history をページベースで呼び出し、1ページ分の messages を yield するジェネレータ。"""
```
[the remark this answers is not in the document]
Comment (hihats) [reply slack_conversations.py:26]: @copilot Make issue

File: slack_rate_limit.py [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+def handle_rate_limit(func, *args, max_retries=5, base_delay=1, **kwargs):
+    """レート制限に対応するためのラッパー関数
+
+    Retry-After ヘッダーがある場合はその値を使用し、
+    ない場合は base_delay を基にした指数バックオフ (base_delay * 2^attempt) を適用する。
+    """
+    status_code = None
+    for attempt in range(max_retries):
+        try:
+            return func(*args, **kwargs)
```
[the remark this answers is not in the document]
Comment (hihats) [reply slack_rate_limit.py:22]: @copilot Make issue

File: slack_conversations.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (hihats) [reply slack_conversations.py:26]: `tests/test_slack_conversations.py`（d9dc0ba で追加）で解消済みです。cursor pagination（single/multi/3ページの has_more・next_cursor 連鎖）、skip_parent の挙動（first-page のみ適用・empty first page）、rate-limit retry、ok=False / 空応答までカバーしています。
